From b2a53ed713ede86e7626435806623700b9f75de9 Mon Sep 17 00:00:00 2001
From: Shayne Chen <shayne.chen@mediatek.com>
Date: Mon, 20 Jan 2025 13:53:10 +0800
Subject: [PATCH 079/104] mtk: mt76: mt7996: cleanup mt7996_hw_phy()

In upstream codebase, mt7996_hw_phy() is removed, since it always eqauls
to &dev->phy. Cleanup remaining parts from internal codebase.

Signed-off-by: Shayne Chen <shayne.chen@mediatek.com>
---
 mt7996/main.c   | 16 +++++++---------
 mt7996/mt7996.h |  8 --------
 mt7996/vendor.c | 21 ++++++++++-----------
 3 files changed, 17 insertions(+), 28 deletions(-)

diff --git a/mt7996/main.c b/mt7996/main.c
index a9daccd2..0a719cba 100644
--- a/mt7996/main.c
+++ b/mt7996/main.c
@@ -518,13 +518,13 @@ static int mt7996_add_interface(struct ieee80211_hw *hw,
 	mvif->dev = dev;
 	mvif->sta.vif = mvif;
 	/* TODO: temporaily set this to prevent some crashes */
-	mvif->deflink.phy = mt7996_hw_phy(hw);
+	mvif->deflink.phy = &dev->phy;
 	mvif->deflink.mt76.mvif = &mvif->mt76;
 	memset(mvif->mt76.band_to_link, IEEE80211_LINK_UNSPECIFIED,
 	       sizeof(mvif->mt76.band_to_link));
 
 	if (vif->type == NL80211_IFTYPE_STATION)
-		err = mt7996_vif_link_add(mt7996_hw_phy(hw)->mt76, vif, &vif->bss_conf, NULL);
+		err = mt7996_vif_link_add(mvif->deflink.phy->mt76, vif, &vif->bss_conf, NULL);
 
 out:
 	mutex_unlock(&dev->mt76.mutex);
@@ -554,7 +554,7 @@ static void mt7996_remove_interface(struct ieee80211_hw *hw,
 	struct mt7996_vif *mvif = (struct mt7996_vif *)vif->drv_priv;
 	struct mt7996_vif_link *mconf;
 	struct mt7996_dev *dev = mt7996_hw_dev(hw);
-	struct mt7996_phy *phy = mt7996_hw_phy(hw);
+	struct mt7996_phy *phy = &dev->phy;
 	struct mt7996_radio_data rdata = {};
 	int i;
 
@@ -2048,8 +2048,7 @@ static void mt7996_link_sta_rc_update(struct ieee80211_hw *hw,
 {
 	struct ieee80211_sta *sta = link_sta->sta;
 	struct mt7996_sta *msta = (struct mt7996_sta *)sta->drv_priv;
-	struct mt7996_phy *phy = mt7996_hw_phy(hw);
-	struct mt7996_dev *dev = phy->dev;
+	struct mt7996_dev *dev = mt7996_hw_dev(hw);
 	struct mt7996_sta_rc_work_data data = {
 		.link_id = msta->pri_link,
 		.changed = changed,
@@ -2070,8 +2069,7 @@ mt7996_set_bitrate_mask(struct ieee80211_hw *hw, struct ieee80211_vif *vif,
 			const struct cfg80211_bitrate_mask *mask,
 			unsigned int link_id)
 {
-	struct mt7996_phy *phy = mt7996_hw_phy(hw);
-	struct mt7996_dev *dev = phy->dev;
+	struct mt7996_dev *dev = mt7996_hw_dev(hw);
 	struct mt7996_vif_link *mconf;
 	struct mt7996_sta_rc_work_data data = {
 		.link_id = link_id,
@@ -2315,7 +2313,7 @@ void mt7996_get_et_stats(struct ieee80211_hw *hw,
 			 struct ethtool_stats *stats, u64 *data)
 {
 	struct mt7996_dev *dev = mt7996_hw_dev(hw);
-	struct mt7996_phy *phy = mt7996_hw_phy(hw);
+	struct mt7996_phy *phy = &dev->phy;
 	struct mt7996_vif *mvif = (struct mt7996_vif *)vif->drv_priv;
 	struct mt7996_vif_link *mconf;
 	struct mt76_mib_stats *mib = &phy->mib;
@@ -2599,7 +2597,7 @@ mt7996_change_vif_links(struct ieee80211_hw *hw, struct ieee80211_vif *vif,
 			struct ieee80211_bss_conf *old[IEEE80211_MLD_MAX_NUM_LINKS])
 {
 	struct mt7996_dev *dev = mt7996_hw_dev(hw);
-	struct mt7996_phy *phy = mt7996_hw_phy(hw);
+	struct mt7996_phy *phy = &dev->phy;
 	struct mt7996_vif *mvif = (struct mt7996_vif *)vif->drv_priv;
 	struct mt7996_vif_link *mconf;
 	unsigned long rem = old_links & ~new_links & ~vif->dormant_links;
diff --git a/mt7996/mt7996.h b/mt7996/mt7996.h
index 2ce053d7..8fa72d85 100644
--- a/mt7996/mt7996.h
+++ b/mt7996/mt7996.h
@@ -974,14 +974,6 @@ enum mt7996_rdd_cmd {
 
 #define RDD_ZW_TIMER_OFF	BIT(31)
 
-static inline struct mt7996_phy *
-mt7996_hw_phy(struct ieee80211_hw *hw)
-{
-	struct mt76_phy *phy = hw->priv;
-
-	return phy->priv;
-}
-
 static inline struct mt7996_dev *
 mt7996_hw_dev(struct ieee80211_hw *hw)
 {
diff --git a/mt7996/vendor.c b/mt7996/vendor.c
index ae40f05d..07ee7afc 100644
--- a/mt7996/vendor.c
+++ b/mt7996/vendor.c
@@ -260,7 +260,8 @@ error:
 }
 
 void mt7996_set_wireless_rts_sigta(struct ieee80211_hw *hw, u8 value) {
-	struct mt7996_phy *phy = mt7996_hw_phy(hw);
+	struct mt7996_dev *dev = mt7996_hw_dev(hw);
+	struct mt7996_phy *phy = &dev->phy;
 
 	switch (value) {
 	case BW_SIGNALING_STATIC:
@@ -683,7 +684,8 @@ static int mt7996_vendor_edcca_ctrl(struct wiphy *wiphy, struct wireless_dev *wd
 				    const void *data, int data_len)
 {
 	struct ieee80211_hw *hw = wiphy_to_ieee80211_hw(wiphy);
-	struct mt7996_phy *phy = mt7996_hw_phy(hw);
+	struct mt7996_dev *dev = mt7996_hw_dev(hw);
+	struct mt7996_phy *phy = &dev->phy;
 	struct nlattr *tb[NUM_MTK_VENDOR_ATTRS_EDCCA_CTRL];
 	int err;
 	u8 edcca_mode;
@@ -742,7 +744,8 @@ mt7996_vendor_edcca_ctrl_dump(struct wiphy *wiphy, struct wireless_dev *wdev,
 			     unsigned long *storage)
 {
 	struct ieee80211_hw *hw = wiphy_to_ieee80211_hw(wiphy);
-	struct mt7996_phy *phy = mt7996_hw_phy(hw);
+	struct mt7996_dev *dev = mt7996_hw_dev(hw);
+	struct mt7996_phy *phy = &dev->phy;
 	struct nlattr *tb[NUM_MTK_VENDOR_ATTRS_EDCCA_CTRL];
 	int err;
 	u8 edcca_mode;
@@ -818,8 +821,7 @@ static int mt7996_vendor_ibf_ctrl(struct wiphy *wiphy,
 				  int data_len)
 {
 	struct ieee80211_hw *hw = wiphy_to_ieee80211_hw(wiphy);
-	struct mt7996_phy *phy = mt7996_hw_phy(hw);
-	struct mt7996_dev *dev = phy->dev;
+	struct mt7996_dev *dev = mt7996_hw_dev(hw);
 	struct nlattr *tb[NUM_MTK_VENDOR_ATTRS_IBF_CTRL];
 	int err;
 	u8 val;
@@ -847,8 +849,7 @@ mt7996_vendor_ibf_ctrl_dump(struct wiphy *wiphy, struct wireless_dev *wdev,
 			    unsigned long *storage)
 {
 	struct ieee80211_hw *hw = wiphy_to_ieee80211_hw(wiphy);
-	struct mt7996_phy *phy = mt7996_hw_phy(hw);
-	struct mt7996_dev *dev = phy->dev;
+	struct mt7996_dev *dev = mt7996_hw_dev(hw);
 
 	if (*storage == 1)
 		return -ENOENT;
@@ -971,8 +972,7 @@ static int mt7996_vendor_rfeature_ctrl(struct wiphy *wiphy,
 				       int data_len)
 {
 	struct ieee80211_hw *hw = wiphy_to_ieee80211_hw(wiphy);
-	struct mt7996_phy *phy = mt7996_hw_phy(hw);
-	struct mt7996_dev *dev = phy->dev;
+	struct mt7996_dev *dev = mt7996_hw_dev(hw);
 	struct ieee80211_vif *vif = wdev_to_ieee80211_vif(wdev);
 	struct mt7996_vif *mvif = (struct mt7996_vif *)vif->drv_priv;
 	struct mt7996_vif_link *mconf;
@@ -1057,8 +1057,7 @@ static int mt7996_vendor_wireless_ctrl(struct wiphy *wiphy,
 				       int data_len)
 {
 	struct ieee80211_hw *hw = wiphy_to_ieee80211_hw(wiphy);
-	struct mt7996_phy *phy = mt7996_hw_phy(hw);
-	struct mt7996_dev *dev = phy->dev;
+	struct mt7996_dev *dev = mt7996_hw_dev(hw);
 	struct ieee80211_vif *vif = wdev_to_ieee80211_vif(wdev);
 	struct mt7996_vif *mvif = (struct mt7996_vif *)vif->drv_priv;
 	struct mt7996_vif_link *mconf;
-- 
2.45.2

